winpr/libwinpr/sspi/Kerberos/kerberos.c

```diff
@@ -320,73 +305,40 @@ static SECURITY_STATUS SEC_ENTRY kerberos_AcquireCredentialsHandleA(
 		{
 			if (krb_settings->startTime)
 				start_time = krb_settings->startTime;
-			if (krb_settings->lifeTime)
-				krb5_get_init_creds_opt_set_tkt_life(gic_opt, krb_settings->lifeTime);
-			if (krb_settings->renewLifeTime)
-				krb5_get_init_creds_opt_set_renew_life(gic_opt, krb_settings->renewLifeTime);
-			if (krb_settings->withPac)
-				krb5_get_init_creds_opt_set_pac_request(ctx, gic_opt, TRUE);
-			if (krb_settings->armorCache)
-				krb5_get_init_creds_opt_set_fast_ccache_name(ctx, gic_opt,
-				                                             krb_settings->armorCache);
-			if (krb_settings->pkinitX509Identity)
-				krb5_get_init_creds_opt_set_pa(ctx, gic_opt, "X509_user_identity",
-				                               krb_settings->pkinitX509Identity);
-			if (krb_settings->pkinitX509Anchors)
-				krb5_get_init_creds_opt_set_pa(ctx, gic_opt, "X509_anchors",
-				                               krb_settings->pkinitX509Anchors);
+			kerberos_set_krb_opts(ctx, gic_opt, krb_settings);
 		}
 
 #ifdef WITH_GSSAPI_MIT
 		krb5_get_init_creds_opt_set_out_ccache(ctx, gic_opt, ccache);
 		krb5_get_init_creds_opt_set_in_ccache(ctx, gic_opt, ccache);
 #endif
-	}
-
-	krb5_cc_get_full_name(ctx, ccache, &ccache_name);
-	krb5_kt_get_name(ctx, keytab, keytab_name, PATH_MAX);
-
-	cred_store_opts[0].key = "ccache";
-	cred_store_opts[0].value = ccache_name;
-	cred_store_opts[1].key = "keytab";
-	cred_store_opts[1].value = keytab_name;
-
-	/* Check if there are initial creds already in the cache there's no need to request new ones */
-	major = sspi_gss_acquire_cred_from(&minor, SSPI_GSS_C_NO_NAME, SSPI_GSS_C_INDEFINITE,
-	                                   &desired_mechs, cred_usage, &cred_store, &gss_creds, NULL,
-	                                   &time_rec);
-	gss_log_status_messages(major, minor);
 
-	/* No use getting initial creds for inbound creds */
-	if (!(fCredentialUse & SECPKG_CRED_OUTBOUND))
-		goto cleanup;
-
-	if (!sspi_is_no_creds(major, minor) && time_rec > 0)
-		goto cleanup;
+		if ((rv = krb5_init_creds_init(ctx, principal, krb5_prompter, password, start_time, gic_opt,
+		                               &creds_ctx)))
+			goto cleanup;
 
-	if ((rv = krb5_get_init_creds_password(ctx, &creds, principal, password, krb5_prompter,
-	                                       password, start_time, NULL, gic_opt)))
-		goto cleanup;
+		if ((rv = krb5_init_creds_get(ctx, creds_ctx)))
+			goto cleanup;
 
 #ifdef WITH_GSSAPI_HEIMDAL
-	if (rv = krb5_cc_store_cred(ctx, ccache, &creds))
-		goto cleanup;
+		krb5_creds creds;
```

Comment:
This variable shall be declared at the start of the block or artificially create a block around these statements